pkg/controller/status/controller.go

```diff
@@ -166,7 +166,7 @@ func (c *Controller) merge(clusterOperator *configv1.ClusterOperator) *configv1.
 }
 
 // calculate the current controller status based on its given sources
-func (c *Controller) currentControllerStatus() (allReady bool, lastTransition time.Time) {
+func (c *Controller) currentControllerStatus() (allReady bool, lastTransition time.Time) { //nolint: gocyclo
```

Comment:
Maybe it is not the goal here... but since we have an cyclomatic complexities alert being ignored here would be a good idea to handle every `if/if-else` into separated functions, and to simplify it with an `switch-case`. What do you think?

Reply:
Yes, but not as a part of this task/PR.